core/src/main/kotlin/com/justai/jaicf/channel/http/Ktor.kt

```diff
@@ -43,14 +44,14 @@ fun Routing.httpBotRouting(vararg channels: Pair<String, HttpBotChannel>) {
             val response = withContext(Dispatchers.Default) {
                 channel.second.process(request)
             }
-            response?.headers?.forEach { call.response.headers.append(it.key, it.value, false) }
+            response.headers.forEach { call.response.headers.append(it.key, it.value, false) }
 
-            when (response) {
-                null -> call.respond(HttpStatusCode.NotFound, "Bot didn't respond")
-                else -> call.respondOutputStream(ContentType.parse(response.contentType)) {
-                    response.output.writeTo(this)
-                }
+            call.respondOutputStream(
+                ContentType.parse(response.contentType.value),
+                HttpStatusCode.fromValue(response.statusCode.value)
+            ) {
+                response.output.writeTo(this)
```

Comment:
Could you please check if this blocking code causes any errors on ktor threads? Shouldn't we move it to another coroutine context?

Reply:
Ktor itself runs it in the IO context.